src/auth/oauth-handler.ts

```diff
@@ -0,0 +1,81 @@
+import { gql } from 'graphql-request'
+
+import { AuthSessionCookie } from './cookie/auth-session-cookie'
+import { createAuthAwareGraphqlFetch } from '@/api/graphql-fetch'
+
+type HandlerType = 'login' | 'logout' | 'consent'
+
+const queries: Record<HandlerType, string> = {
+  login: gql`
+    mutation ($input: OauthAcceptInput!) {
+      oauth {
+        acceptLogin(input: $input) {
+          redirectUri
+        }
+      }
+    }
+  `,
+  consent: gql`
+    mutation ($input: OauthAcceptInput!) {
+      oauth {
+        acceptConsent(input: $input) {
+          redirectUri
+        }
+      }
+    }
+  `,
+  logout: gql`
+    mutation ($challenge: String!) {
+      oauth {
+        acceptLogout(challenge: $challenge) {
+          redirectUri
+        }
+      }
+    }
+  `,
+}
+
+const throwError = (error?: unknown) => {
+  // eslint-disable-next-line no-console
+  console.error(error)
+  throw new Error('problem in oauth')
+}
+
+export const oauthHandler = async (type: HandlerType, challenge?: string) => {
+  const session = AuthSessionCookie.parse()
+
+  if (!challenge) throwError('no challenge provided!')
+
+  const variables = {
+    input: {
+      session,
+      challenge,
+    },
+  }
+  const args = JSON.stringify({ query: queries[type], variables })
+
+  try {
+    const dummyAuth = { username: '', id: 0 }
```

Comment:
Shouldn't we put a TODO here (or something else that Actions don't complain about, but to remember us to improve it later)?